FordLabs/retroquest
# Retro Timer

## Overview
This recreates the retro timer created in angular [here](https://github.com/FordLabs/retroquest/pull/350) that never got merged in before the react rewrite.

Connects #349 

### Demo
<img width="359" alt="Screen Shot 2022-04-19 at 9 43 16 AM" src="https://user-images.githubusercontent.com/17144844/164017542-6e1196a4-2910-46f4-9fa0-c46a499db893.png">
<img width="383" alt="Screen Shot 2022-04-19 at 9 43 27 AM" src="https://user-images.githubusercontent.com/17144844/164017548-b0070cf9-8f23-4d71-8d0b-1be8cf641f49.png">
<img width="640" alt="Screen Shot 2022-04-14 at 2 02 57 PM" src="https://user-images.githubusercontent.com/17144844/163450953-e22fbc45-11a9-4416-badb-5d539a88ca15.png">

### Notes
* The timer is hidden on mobile

## Testing Instructions
1) Log in and try out the timer. Pick a time (30seconds, 1min, 5min, or 10min) and click play.
2) Ensure the timer starts and counts down fully.
3) When timer gets to 0, ensure that a modal pops up saying that the time is up. 
4) Clicking `Okay!` on the modal should reset the timer.
5) Clicking `Add 1 Minute` should add another minute to the timer and start the timer.
6) Ensure timer and modal looks good in both light and dark mode.
